athenz/src/main/java/com/linecorp/armeria/server/athenz/AbstractAthenzServiceBuilder.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2025 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.athenz;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+
+import com.yahoo.athenz.zpe.ZpeClient;
+import com.yahoo.athenz.zpe.pkey.PublicKeyStore;
+
+import com.linecorp.armeria.client.athenz.ZtsBaseClient;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A base builder for creating an Athenz service that checks access permissions using Athenz policies.
+ */
+@UnstableApi
+public abstract class AbstractAthenzServiceBuilder<SELF extends AbstractAthenzServiceBuilder<SELF>> {
+
+    private static final Duration DEFAULT_OAUTH2_KEYS_REFRESH_INTERVAL = Duration.ofHours(1);
+    private static final int MAX_TOKEN_CACHE_SIZE = 10240;
+
+    private Duration oauth2KeysRefreshInterval = DEFAULT_OAUTH2_KEYS_REFRESH_INTERVAL;
+
+    private final ZtsBaseClient ztsBaseClient;
+    @Nullable
+    private AthenzPolicyConfig policyConfig;
+    private int maxTokenCacheSize = MAX_TOKEN_CACHE_SIZE;
+
+    AbstractAthenzServiceBuilder(ZtsBaseClient ztsBaseClient) {
```

Comment:
> The external zpu cronjob to fetch policies is no longer necessary.

I actually understood the zpu cronjob is still necessary since `AthenzService` still needs to use the credentials created/rotated by the zpu when connecting to zts. Let me know if I misunderstood.

Reply:
ZPU fetches and refreshes Athenz policy files. IIUC, it is not related to credentials. Generally, creating Service credentials is delegated to Athenz SIA, which is still necessary.
 https://github.com/AthenZ/athenz/blob/master/docs/system_view.md#sia-service-identity-agent-provider:~:text=for%20issuing%20tokens.-,SIA%20(Service%20Identity%20Agent,can%20validate%20the%20signature.,-ZPE%20(AuthZ%20Policy

Reply:
I see.  I still understood that service providers will need to install either an SIA or ZPU when using this module.

Reply:
This Athenz module does not support creating Service credentials, so an external system such as SIA is still used for mTLS.